sdk/translation/ai-translation-document-rest/test/public/utils/recordedClient.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Context } from "mocha";
+import { Recorder, env } from "@azure-tools/test-recorder";
+import { ClientOptions } from "@azure-rest/core-client";
+import { DocumentTranslationClient } from "../../../src";
+import createClient from "../../../src/documentTranslationClient";
+import { KeyCredential, TokenCredential } from "@azure/core-auth";
+
+export async function startRecorder(context: Context): Promise<Recorder> {
+  const recorder = new Recorder(context.currentTest);
+  await recorder.start({
+    envSetupForPlayback: {
+      DOCUMENT_TRANSLATION_API_KEY: "fakeApiKey",
+      DOCUMENT_TRANSLATION_ENDPOINT:
+        "https://fakeEndpoint-doctranslation.cognitiveservices.azure.com",
+      DOCUMENT_TRANSLATION_STORAGE_NAME: "fakestoragename",
+      DOCUMENT_TRANSLATION_CONNECTION_STRING:
+        "DefaultEndpointsProtocol=https;AccountName=fakeStorageName;AccountKey=fakeKey;EndpointSuffix=core.windows.net",
```

Comment:
cc @harsha-nalluru @timovv - how should these tests adapt to use federated credentials?

Reply:
Will follow up with @harsha-nalluru @timovv later. Resolving this for now, so I can merge this PR